### PR TITLE
feat(nvme): add allowed-hosts control for nvme targets

### PR DESCRIPTION
**Pull request by tiagolobocastro**
This allows us to control which hosts (nqn's) are allowed to connect to our nvmf targets. In addition, we also disconnect previously allowed devices which are no longer allowed.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
Same old: Address already in use (98)
bors try

File: io-engine/src/bin/io-engine-client/bdev_cli.rs [hunk 1]
```diff
@@ -49,6 +49,16 @@ pub fn subcommands<'a, 'b>() -> App<'a, 'b> {
...
                 .default_value("nvmf"),
+        )
+        .arg(
+            Arg::with_name("allowed-host")
+                .long("allowed-host")
+                .takes_value(true)
+                .multiple(true)
+                .required(false)
+                .help(
+                    "NQN's of hosts which are allowed to connect to the target",
```
Comment (dsavitskiy) [inline io-engine/src/bin/io-engine-client/bdev_cli.rs]: NQN's: hmm, maybe apostrophe is not need here?

**Review by dsavitskiy — APPROVED** [review]

File: io-engine/src/bin/io-engine-client/bdev_cli.rs (hunk 1 above)
Comment (tiagolobocastro) [reply io-engine/src/bin/io-engine-client/bdev_cli.rs]: yes I don't think so, I'll amend it.

File: io-engine/src/bdev/nexus/nexus_share.rs [hunk 2]
```diff
@@ -86,9 +98,18 @@ impl From<&NexusTarget> for Protocol {
 impl<'n> Nexus<'n> {
     /// TODO
     pub async fn share(
+        self: Pin<&mut Self>,
+        protocol: Protocol,
+        key: Option<String>,
+    ) -> Result<String, Error> {
+        self.share_ext(protocol, key, vec![]).await
+    }
+    /// TODO
```
Comment (Abhinandan-Purkait) [inline io-engine/src/bdev/nexus/nexus_share.rs:107]: Are these TODOs for adding doc comments?

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

File: io-engine/src/bdev/nexus/nexus_share.rs (hunk 2 above)
Comment (tiagolobocastro) [reply io-engine/src/bdev/nexus/nexus_share.rs:107]: Not sure, maybe. Left it like that as it was already as such..

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
I've added a commit to clean previous runs of cargo-test, let's see if this makes CI more reliable @mtzaurus 
bors merge

**Review by mtzaurus — APPROVED** [review]

**Pull request merged**